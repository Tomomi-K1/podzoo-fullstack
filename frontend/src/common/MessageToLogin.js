import * as React from "react";
// Material UI
import Typography from "@mui/material/Typography";
import Button from "@mui/material/Button";
import Popper from "@mui/material/Popper";
import PopupState, { bindToggle, bindPopper } from "material-ui-popup-state";
import Fade from "@mui/material/Fade";
import Paper from "@mui/material/Paper";
import FavoriteBorderIcon from '@mui/icons-material/FavoriteBorder';

/** MessageToLogin : Button
 * shows button that has pop up message to tell user 
   that you need to login to use favorite or write reivews.
   This component is used when there is no logged in user
 * it accepts message prop 
 * message prop is used as a word on Button
 * component using MessageToLogin : PodcastDetailedLayout.js
 */
function MessageToLogin({message}) {
  return (
    <PopupState variant="popper" popupId="request-login-popper">
      {(popupState) => (
        <div>
          {/* if message is 'favorite', we are adding FavoriteIcon otherwise we will just show the message prop on the button*/}
          {message==='favorite'
            ? (
                <Button variant="default" {...bindToggle(popupState)}>
                  <Typography variant='body1' sx={{display:'inline-block', mx:1}}>Favorite</Typography>
                  <FavoriteBorderIcon />   
                </Button>
              )
            :( 
                <Button variant="default" {...bindToggle(popupState)}>
                  {message}
                </Button>
              )
          }
          <Popper {...bindPopper(popupState)} transition>
            {({ TransitionProps }) => (
              <Fade {...TransitionProps} timeout={350}>
                <Paper>
                  <Typography sx={{ p: 2 }}>
                    You need to login/signup before accessing this feature.
                  </Typography>
                </Paper>
              </Fade>
            )}
          </Popper>
        </div>
      )}
    </PopupState>
  );
}

export default MessageToLogin;