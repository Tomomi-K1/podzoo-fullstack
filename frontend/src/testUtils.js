import React from "react";
import UserContext from "./UserContext";

const demoUser = {
  username: "testuser",
  email: "test@test.net",
  fav:[1,2,3]
};

const favoritesArray =[1, 2]

const UserProvider =
    ({ children, currentUser = demoUser, checkFavPod=()=>true, favorites= favoritesArray}) => (
    <UserContext.Provider value={{ currentUser, checkFavPod, favorites }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };