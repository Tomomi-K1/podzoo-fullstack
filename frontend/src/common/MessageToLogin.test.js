import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import MessageToLogin from "./MessageToLogin";
import { UserProvider } from "../testUtils";

it('renders favorite button', () => {
    const message = 'favorite'
    render(
        <MemoryRouter>
            <UserProvider currentUser={null}>
                <MessageToLogin message={message} />
            </UserProvider>
        </MemoryRouter>
    ) 
})

it('renders button with message prop on it', () => {
    const message = 'Write Your Review'
    render(
        <MemoryRouter>
            <UserProvider currentUser={null}>
                <MessageToLogin message={message} />
            </UserProvider>
        </MemoryRouter>
    )
    expect(screen.getByText('Write Your Review')).toBeInTheDocument(); 
})

it('matches snapshot', () => {
    const message = 'Write Your Review'
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider currentUser={null}>
                <MessageToLogin message={message} />
            </UserProvider>
        </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot(); 
})
