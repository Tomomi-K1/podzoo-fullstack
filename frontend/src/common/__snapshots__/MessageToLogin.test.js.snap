// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-default MuiButton-defaultPrimary MuiButton-sizeMedium MuiButton-defaultSizeMedium MuiButton-root MuiButton-default MuiButton-defaultPrimary MuiButton-sizeMedium MuiButton-defaultSizeMedium css-1w1rijm-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Write Your Review
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
